frontend/main.js

```diff
@@ -309,6 +309,12 @@ async function startApp () {
           },
           'reconnection-succeeded' () {
             sbp('gi.ui/clearBanner')
+          },
+          'subscription-succeeded' (event) {
+            const { channelID } = event.detail
+            if (channelID in sbp('state/vuex/state').contracts && !sbp('chelonia/contract/isSyncing', channelID)) {
```

Comment:
Because of how `/sync` works now, calling `/isSyncing` first is unnecessary (it should be a no-op in this case). Also, since `/sync` is asynchronous, there should be a `.catch` at the end to handle errors gracefully.

Reply:
Thanks!

Reply:
@snowteamer I remember vaguely that there was an event that caused a subscribe to happen when a reconnection was established on the websocket (after say the internet was out for a bit). Is this that event?

Because if so we need to force it to fetch the latest events with `{ force: true }` I think...

@snowteamer & @corrideat can you investigate this question?

Reply:
@taoeffect That sounds like the `reconnection-succeeded` event, which causes pending `sub` actions to be retried. Therefore many subscriptions may be triggered upon reconnection.
I thought fetching latest events was the default behavior of `sync`.

Reply:
@snowteamer `sync` now takes an extra parameter (an options object). Have a look at how it's implemented. The reason `force` was added was because `/sync` is called a *lot* of times in various places.

Reply:
@taoeffect So that was added to avoid unnecessary network requests, e.g. to avoid checking the contract's `latestHash` every time `/sync `is called? Ok, that makes sense.

But I'm not sure why a plain `/sync` call couldn't automatically bail out without network calls if the system knows that the given contract is already fully synced.

In my understanding, a contract is already synced unless:
- it is new, i.e. we have no log entries for it yet ;
- it is marked as currently syncing ;
- we just logged in or reconnected after a disconnection, hence we might have missed new log entries in the meantime which should be fetched.